### PR TITLE
wagmi wallet integration

### PR DESCRIPTION
**Pull request by antho1404**
### Description

Integrate wagmi as wallet provider instead of the previous web3 react.
Added better management of user's connection to only be connected on the platform only when the wallet is connected AND the wallet is logged in to the API thanks to the `useAccount` hook.

### How to test

Connect the wallet on the platform on the login page or other and check that every feature work as expected.
Test when you cleanup the cookies (you should be asked to sign again)
Test to disconnect your wallet
Test to logout on the platform
Test to be on the wrong network
...

### Known issues

~~When refreshing from some pages (like the explore page) while being connected, the `onConnect` from wagmi is not triggered making the login flow "stuck"
Still debugging this part~~ Done thanks to @ismailToyran

### CODE REVIEW

**Comment by EmmanuelDrouin** [comment]
[Implement Wagmi](https://trello.com/c/zOsJhrwy/423-implement-wagmi)

File: hooks/useEagerConnect.ts [hunk 1]
```diff
@@ -1,66 +1,23 @@
...
   useEffect(() => {
-    if (active) {
-      setTried(true)
-    }
-  }, [active])
+    if (ready) setHasBeenReady(true)
+  }, [ready, setHasBeenReady])
 
-  return tried
+  console.log({ ready: hasBeenReady || ready })
```
Comment (ismailToyran) [inline hooks/useEagerConnect.ts:21]: Kept this `console.log` on purpose?

File: hooks/useEagerConnect.ts [hunk 2]
```diff
@@ -1,66 +1,23 @@
...
-        if (isAuthorized) {
-          const address = await connectors.injected.getAccount()
-          if (address?.toLowerCase() === connectedAddress?.toLowerCase()) {
-            void activate(connectors.injected, undefined, true)
-            return
-          }
-        }
+  const ready = useMemo(() => {
+    if (isReconnecting) return false
+    if (!isConnected) return true
```
Comment (ismailToyran) [inline hooks/useEagerConnect.ts:12]: Doesn't this one needs to be `false`?

File: components/Login/Form.tsx [hunk 3]
```diff
@@ -1,39 +1,41 @@
...
 import WalletWalletConnect from '../Wallet/Connectors/WalletConnect'
 
 type Props = {
-  activate: (
-    connector: AbstractConnector,
-    onError?: (error: Error) => void,
-    throwErrors?: boolean,
-  ) => Promise<void>
-  networkName: string
+  onActivate?: () => void
```
Comment (antho1404) [inline components/Login/Form.tsx:11]: This now becomes a simple callback for "notification" and not to actually activate

File: components/Login/Form.tsx [hunk 4]
```diff
@@ -1,39 +1,41 @@
...
-  )
 
   const hasStandardWallet =
     connectors.injected || connectors.coinbase || connectors.walletConnect
 
   return (
     <>
-      {connectors.email && <WalletEmail activate={activate} />}
+      {connectors.email && (
+        <Stack
```
Comment (antho1404) [inline components/Login/Form.tsx:25]: Kept a similar design as the buttons as this library automatically opens the popup from magic.

File: pages/_app.tsx [hunk 5]
```diff
@@ -167,65 +158,50 @@ function AccountProvider(
...
             }
           : {},
-        cache: new InMemoryCache({
-          typePolicies: {
-            Account: {
-              keyFields: ['address'],
-            },
-          },
-        }).restore(props.cache),
+        cache: new InMemoryCache({}).restore(props.cache),
```
Comment (ismailToyran) [inline pages/_app.tsx:199]: Compared to Admin panel we are missing
```ts
cache: new InMemoryCache({
  typePolicies: {
    Account: {
      keyFields: ['address'],
    },
  },
}).restore(props.cache),
```
Any reason to include on admin and not this one?

**Review by ismailToyran — COMMENTED** [review]
Everything works great.
I've seen you've added new library for email connection, tried that one as well and that works great as well. I'll approve when you provide me some context on the conversations I've created.

File: hooks/useEagerConnect.ts (hunk 1 above)
Comment (antho1404) [reply hooks/useEagerConnect.ts]: just removed

File: hooks/useEagerConnect.ts (hunk 2 above)
Comment (antho1404) [reply hooks/useEagerConnect.ts:12]: No, but indeed it's weird, here, if we are not connected (aka the wallet connected), then it means we tried we don't need to try more. This boolean here is either we tried or not. When we know we tried, it means it's ready (even if the result means that we are not connected)

File: pages/_app.tsx (hunk 5 above)
Comment (antho1404) [reply pages/_app.tsx]: Good catch, that was an error.
Just reverted

File: hooks/useSigner.ts [hunk 6]
```diff
@@ -1,18 +1,20 @@
...
+  const { isLoggedIn } = useAccount()
 
   return useMemo(() => {
-    if (!library) return
-    if (!account) return
-    return library.getSigner(account)
-  }, [library, account])
+    if (!isLoggedIn) return
+    if (!data) return
+    return (data as Signer & TypedDataSigner) || undefined
```
[the remark this answers is not in the document]
Comment (antho1404) [reply hooks/useSigner.ts:18]: Not really, this is to type with the `TypedDataSigner`, the signer from wagmi is already an implementation of `Signer` from `@ethersproject/abstract-signer` so all good here

File: package.json [hunk 7]
```diff
@@ -17,22 +17,17 @@
     "@apollo/client": "^3.5.9",
     "@bugsnag/js": "^7.18.0",
     "@bugsnag/plugin-react": "^7.18.0",
-    "@chakra-ui/react": "^1.8.8",
+    "@chakra-ui/react": "^1.8.9",
```
[the remark this answers is not in the document]
Comment (ismailToyran) [reply package.json:20]: Great find 🎉

**Review by ismailToyran — APPROVED** [review]
Working as intended 👍 
Make sure to check my commits before merge though.

**Pull request merged**